src/postOffice/MultichainPostman.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.8.16;
+
+import { CallProxy } from "../interfaces/adapters/IMultichainAdapter.sol";
+import { IPostOffice } from "../interfaces/postOffice/IPostOffice.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/MsgStructs.sol";
+
+contract MultichainPostman is Ownable {
+	address public anyCall;
+	address public anycallExecutor;
+
+	IPostOffice public immutable postOffice;
```

Comment:
(same for LayerZeroPostman)

Reply:
I'm guessing its for simplifying reading messages arriving to XVault from different adapters... We could handle this by looping through the chain-vault struct and checking on each adapter. 

On the chain-vault side we could do something similar - store an array of x-chain vaults w associated adapters?

Not sure if this is the right way, but wan to make sure we consider this approach.

Reply:
I've been thinking about it and still not clear what is the best solution for this. I'll try to summarise our thoughts about it but we need your help to define the final solution.

Our initial problem: we need to storage messages in one place at least for each Vault. Why? Because different postmen will write message on that board, so the adapter itself cannot hold the message board, we would have more than one board for each vault in different contracts (hard to read and keep track).

Our solution right now: create postOffice which holds the message board for all vaults in a chain. With the postmen being the adapters which holds the logic to write messages on board (receive) and deliver messages to other chains (send). This guarantees 2 things: modularity and all messages are in one place. Modularity because we can always add another postman with their specific logic to send and receiver messages (opens the possibility to integrate more services or update the logic for the current ones). And the board lives on the same place. All logic to read and write messages lives in postOffice as the interface that the vault uses to send messages.

Problems with this solution: for the vault to read messages and take action we need to loop through the postOffice messages and then loop again on the vault to take appropriate action for each message. There're 2 loops on the same array happing. This is true for all message types except Harvest which only the total value is relevant. This makes operational costs higher for manager. Another issue is the access control for the postOffice, right now the vault itself holds which vaults it can send messages and bridge assets and the postOffice holds which vaults and postmen it can send and receive messages from.

Alternative approach: split postOffice logic into postmen and vaults. The message board would live in vault as the send and write message logic. Also, the access control would live only in vault with a whitelist of vaults and postmen that it can interact with. This would save us the double loop in messages array. The problem is that the vault code would increase a little bit (I don't believe we would reach the limit yet) and if we deploy a new postman we need to go on each vault and whitelist/update the postmen list.

After some thoughts I'm inclined to change the code and split the postOffice but this means delaying the project even more because we also would need to change deploy scripts and the tests that we have implemented (3 hours and this is done).

Reply:
I'll think about this some more, but it seems to me that having the postOffice logic live inside the vault makes the most sense unless we're getting close to the size limit.